velox/type/tests/TimestampTest.cpp

```diff
@@ -299,37 +299,152 @@ TEST(TimestampTest, outOfRange) {
       t.toTimezone(*timezone), "Timestamp is outside of supported range");
 }
 
-TEST(TimestampTest, epochToUtc) {
-  std::tm tm;
-  ASSERT_FALSE(epochToUtc(-(1ll << 60), tm));
-  ASSERT_FALSE(epochToUtc(1ll << 60, tm));
+void checkTm(std::tm& actual, std::tm& expected) {
+  ASSERT_EQ(expected.tm_year, actual.tm_year);
+  ASSERT_EQ(expected.tm_yday, actual.tm_yday);
+  ASSERT_EQ(expected.tm_mon, actual.tm_mon);
+  ASSERT_EQ(expected.tm_mday, actual.tm_mday);
+  ASSERT_EQ(expected.tm_wday, actual.tm_wday);
+  ASSERT_EQ(expected.tm_hour, actual.tm_hour);
+  ASSERT_EQ(expected.tm_min, actual.tm_min);
+  ASSERT_EQ(expected.tm_sec, actual.tm_sec);
 }
 
-TEST(TimestampTest, randomEpochToUtc) {
+std::string tmToString(
```

Comment:
Why don't we use this implementation in production code? CC: @Yuhta

Reply:
The `tmToString` here is to compare with the result of `Timestamp::tmToString` to ensure that the time string output by the velox version is consistent with the result output by `std::put_time`.

Reply:
@Yuhta @wypb If std::put_time produces the desired result, then why do we need custom code to produce the same?

Reply:
Because there are performance issues with `std::put_time`,  see [here](https://github.com/facebookincubator/velox/pull/6801).

Reply:
Got it. Thanks.